ecommerce/serializers.py

```diff
@@ -56,3 +60,49 @@ class Meta:
             'price',
             'description',
         )
+
+
+class CheckoutDataSerializer(serializers.ModelSerializer):
+    """Serializer for ecommerce information for a BootcampApplication"""
+    bootcamp_run = BootcampRunSerializer()
+    total_price = serializers.SerializerMethodField()
+    total_paid = serializers.SerializerMethodField()
+    payments = serializers.SerializerMethodField()
+    installments = serializers.SerializerMethodField()
+
+    def get_total_price(self, application):
+        """The personal price for the user, or the full price for the run"""
+        return application.bootcamp_run.personal_price(self.context["request"].user) or Decimal(0)
+
+    def get_total_paid(self, application):
+        """The total paid by the user for this application so far"""
+        return get_total_paid(
+            user=self.context["request"].user,
+            application_id=application.id,
+            run_key=application.bootcamp_run.run_key,
+        )
+
+    def get_payments(self, application):
+        """Serialized payments made by the user"""
+        return LineSerializer(
```

Comment:
I think we need to move away from querying for `Line`s that have a matching key, and instead fetch the `Line` from the `application` foreign key in `Order`. Something like `LineSerializer(instance=[order.lines.first() for order in application.orders.all()`. If a user applies to some bootcamp run multiple times for whatever reason, we would only want to see the payments for the specific application in the request. On top of that, this approach would save us some queries, particularly if we prefetched on the initial queryset.